### PR TITLE
Update experiment mgmt api variants

### PR DESCRIPTION
**Pull request by imousterian**
# Amplitude Developer Docs PR

- rearrange existing sections so it is easier to read documentation
- add new parameters to `/experiments/new` and `flags/new` endpoints
- add the followings endpoints for both flags/:id and experiments/:id
-- PATCH 
-- POST /variants/:variantKey/users
-- POST /variants
-- PATCH /variants/:variantKey
-- DELETE variants/:variantKey
-- DELETE variants/:variantKey/users/:userIndex

## Description

Describe your changes. 

## Deadline

Once we deploy the appropriate backend changes (scheduled for week of 1/23/23).

## Change type

- [ ] Doc bug fix. Fixes #[insert issue number]. Amplitude contributors include Jira issue number. 
- [x] Doc update.
- [x] New documentation.
- [x] Non-documentation related fix or update.

# PR checklist:

- [x] My documentation follows the style guidelines of this project.
- [x] I previewed my documentation on a local server using `mkdocs serve`.
- [x] Running `mkdocs serve` didn't generate any failures.
- [x] I have performed a self-review of my own documentation.

@amplitude-dev-docs
@caseyamp

### CODE REVIEW

**Comment by imousterian** [comment]
I have rearranged some sections as it makes the docs more readable. Let me know if it doesn't match our style guide, and I can revert those changes.

File: docs/experiment/apis/management-api.md [hunk 1]
```diff
@@ -55,6 +88,8 @@ Endpoints that list resources such as `/experiments/list` will only return a lim
 
 ------
 
+## Experiment Endpoints
```
Comment (caseyamp) [inline docs/experiment/apis/management-api.md:91]:
```suggestion
## Experiment endpoints
```

I know it takes getting used to -  we use sentence case in headings

**Review by caseyamp — COMMENTED** [review]
@imousterian looks good enough to me -- I just had one suggestion, but I'll let @bgiori have the final say. Thanks for doing all of this!

I'd also suggest double-checking the preview for the table's column width to make sure some of them don't actually need to be "big-column" instead of "med-big-column".

File: docs/experiment/apis/management-api.md [hunk 2]
```diff
@@ -478,13 +537,784 @@ Update the rollout weights for an experiment.
...
+
+<!-- TODO example response body -->
+
+!!!example "Example cURL"
+    ```bash
+    curl --request POST \
+      --url 'https://management-api.experiment.amplitude.com/flags/<id>/variants/<variantKey>/users' \
+      --header 'Accept: application/json' \
+      --header 'Authorization: Bearer <management-api-key>'
+      --data '{"inclusions":"<[]>"}'
```
Comment (bgiori) [inline docs/experiment/apis/management-api.md:1139]: This is a confusing structure. Is it a string or an array? The table above says array of strings but this seems to indicate a json array encoded as a string.

File: docs/experiment/apis/management-api.md [hunk 3]
```diff
@@ -478,13 +537,784 @@ Update the rollout weights for an experiment.
...
+|---|----|
+|`id`| Required. String. The flag's ID.|
+
+### Request body
+
+|<div class="med-big-column">Name</div>|Requirement|Type|Description|
+|---|---|---|---|
+|`bucketingKey`| Optional | string | The user property to bucket the user by. |
+|`bucketingSalt`| Optional | string | Flag's bucketing salt. |
+|`bucketingUnit`| Optional | string | Flag's bucketing. |
```
Comment (bgiori) [inline docs/experiment/apis/management-api.md:1227]:
```suggestion
|`bucketingUnit`| Optional | string | Flag's bucketing unit represented by a group type from the accounts add-on. Used for group level bucketing. |
```

File: docs/experiment/apis/management-api.md [hunk 4]
```diff
@@ -478,13 +537,784 @@ Update the rollout weights for an experiment.
...
+A successful request returns a `200 OK` response and a JSON object with the flag's details.
+
+<!-- TODO example response body -->
+
+!!!example "Example cURL"
+    ```bash
+    curl --request POST \
+      --url 'https://management-api.experiment.amplitude.com/flags/new' \
+      --header 'Accept: application/json' \
+      --header 'Authorization: Bearer <management-api-key>'
```
Comment (bgiori) [inline docs/experiment/apis/management-api.md:1285]:
```suggestion
      --header 'Authorization: Bearer <management-api-key>' \
```

File: docs/experiment/apis/management-api.md [hunk 5]
```diff
@@ -194,279 +228,304 @@ Fetch a list of all versions for an experiment.
...
 
 !!!example "Example cURL"
     ```bash
     curl --request POST \
-      --url 'https://management-api.experiment.amplitude.com/experiments/new' \
+      --url 'https://management-api.experiment.amplitude.com/experiments/<id>/variants/<variantKey>/users' \
       --header 'Accept: application/json' \
       --header 'Authorization: Bearer <management-api-key>'
-      --data '{"projectId":"<projectId>","key":"<key>"}'
+      --data '{"inclusions":"<[]>"}'
```
[the remark this answers is not in the document]
Comment (imousterian) [reply docs/experiment/apis/management-api.md:454]: intended request body should be 
```
{ 
     inclusions: [id1, id2, id3]
}
```
Sounds like I need to change the description in the table to `JSON object with **inclusions** as key, and array of strings as value`.

File: docs/experiment/apis/management-api.md [hunk 6]
```diff
@@ -478,13 +537,784 @@ Update the rollout weights for an experiment.
...
+|`subprop_type`| Required | string | Type of filter; can be null. One of `USER`, `GROUP`, `EVENT`. |
+|`subprop_value`| Required | string array | Array of values. |
+
+#### `subprop_op`
+
+An string value representing operations on a property value. Possible values are: `is`, `is not`, `contains`, `does not contain`, `less`, `less or equal`, `greater`, `greater or equal`, `glob match`, `glob does not match`
+
+#### `group_by`
+
+The `group_by` field contains these objects. Experiment currently does not use these objects, and users should pass an empty array instead.
```
[the remark this answers is not in the document]
Comment (imousterian) [reply docs/experiment/apis/management-api.md:582]: yeah, I was going back and forth about it. I am not sure why we don't use, and whether we will ever use this field. Either way, probably best to hide it.

File: docs/experiment/apis/management-api.md [hunk 7]
```diff
@@ -478,13 +537,784 @@ Update the rollout weights for an experiment.
...
+|<div class="med-big-column">Name</div>|Requirement|Type|Description|
+|---|---|---|---|
+|`projectId`| Required | string | The project's ID. |
+|`key`| Required | string | The experiment key. |
+|`name`| Optional | string | The experiment name. |
+|`description`| Optional | string | Description for the experiment.|
+|`variants`| Optional | object array | Array of [`variants`](#variants). |
+|`bucketingKey`| Optional | string | The user property to bucket the user by. |
+|`rolloutWeights`| Optional | object | Rollout weights for non-targeted users. The object should be a mapping from variant key to rollout weight as an integer. For example: `{ "control": 1, "treatment": 1 }`. |
+|`targetSegments`| Optional | object | See the [`targetSegments`](#targetsegments) table for more information. |
```
[the remark this answers is not in the document]
Comment (imousterian) [reply docs/experiment/apis/management-api.md:626]: right now target segments can only be set on creation. I believe the plan is to change those to use a separate endpoint, `/experiments/<ID>/targeting-rules`, but it is not a part of this change.

File: docs/experiment/apis/management-api.md [hunk 8]
```diff
@@ -194,279 +228,304 @@ Fetch a list of all versions for an experiment.
...
-A successful request returns a `200 OK` response and a JSON object with the experiment's details.
+A successful request returns a `200 OK` response.
 
 <!-- TODO example response body -->
 
 !!!example "Example cURL"
     ```bash
     curl --request POST \
-      --url 'https://management-api.experiment.amplitude.com/experiments/new' \
+      --url 'https://management-api.experiment.amplitude.com/experiments/<id>/variants/<variantKey>/users' \
```
Comment (bgiori) [inline docs/experiment/apis/management-api.md:451]:
```suggestion
      --url 'https://management-api.experiment.amplitude.com/experiments/<id>/variants/<variantKey>/users' \
      --header 'Content-Type: application/json' \
```

File: docs/experiment/apis/management-api.md [hunk 9]
```diff
@@ -478,13 +537,784 @@ Update the rollout weights for an experiment.
...
+A successful request returns a `200 OK` response.
+
+<!-- TODO example response body -->
 
 !!!example "Example cURL"
     ```bash
-    curl --request POST \
-      --url 'https://management-api.experiment.amplitude.com/experiments/<id>/rollout-weights' \
+    curl --request PATCH \
+      --url 'https://management-api.experiment.amplitude.com/experiments/<id>' \
```
Comment (bgiori) [inline docs/experiment/apis/management-api.md:599]:
```suggestion
      --url 'https://management-api.experiment.amplitude.com/experiments/<id>' \
      --header 'Content-Type: application/json' \
```

File: docs/experiment/apis/management-api.md [hunk 10]
```diff
@@ -478,13 +537,784 @@ Update the rollout weights for an experiment.
...
+### Path variables
+
+|<div class="big-column">Name</div>|Description|
+|---|----|
+|`id`| Required. String. The experiment's ID.|
+
+!!!example "Example cURL"
+    ```bash
+    curl --request POST \
+      --url 'https://management-api.experiment.amplitude.com/experiments/<id>/activate' \
```
Comment (bgiori) [inline docs/experiment/apis/management-api.md:706]:
```suggestion
      --url 'https://management-api.experiment.amplitude.com/experiments/<id>/activate' \
      --header 'Content-Type: application/json' \
```

File: docs/experiment/apis/management-api.md [hunk 11]
```diff
@@ -478,13 +537,784 @@ Update the rollout weights for an experiment.
...
+### Path variables
+
+|<div class="big-column">Name</div>|Description|
+|---|----|
+|`id`| Required. String. The experiment's ID.|
+
+!!!example "Example cURL"
+    ```bash
+    curl --request POST \
+      --url 'https://management-api.experiment.amplitude.com/experiments/<id>/deactivate' \
```
Comment (bgiori) [inline docs/experiment/apis/management-api.md:734]:
```suggestion
      --url 'https://management-api.experiment.amplitude.com/experiments/<id>/deactivate' \
      --header 'Content-Type: application/json' \
```

File: docs/experiment/apis/management-api.md [hunk 12]
```diff
@@ -478,13 +537,784 @@ Update the rollout weights for an experiment.
...
+### Request body
+
+|<div class="med-big-column">Name</div>|Requirement|Type|Description|
+|---|---|---|---|
+|`rolloutWeights`| Required | object |A map from variant key to rollout weight. For example:  `{"control": 1,"treatment":1}`. |
+
+!!!example "Example cURL"
+    ```bash
+    curl --request POST \
+      --url 'https://management-api.experiment.amplitude.com/experiments/<id>/rollout-weights' \
```
Comment (bgiori) [inline docs/experiment/apis/management-api.md:768]:
```suggestion
      --url 'https://management-api.experiment.amplitude.com/experiments/<id>/rollout-weights' \
      --header 'Content-Type: application/json' \
```

File: docs/experiment/apis/management-api.md [hunk 13]
```diff
@@ -478,13 +537,784 @@ Update the rollout weights for an experiment.
...
+### Response
+
+A successful request returns a `200 OK` response.
+
+<!-- TODO example response body -->
+
+!!!example "Example cURL"
+    ```bash
+    curl --request POST \
+      --url 'https://management-api.experiment.amplitude.com/flags/<id>/variants' \
```
Comment (bgiori) [inline docs/experiment/apis/management-api.md:1025]:
```suggestion
      --url 'https://management-api.experiment.amplitude.com/flags/<id>/variants' \
      --header 'Content-Type: application/json' \
```

File: docs/experiment/apis/management-api.md [hunk 14]
```diff
@@ -478,13 +537,784 @@ Update the rollout weights for an experiment.
...
+### Response
+
+A successful request returns a `200 OK` response.
+
+<!-- TODO example response body -->
+
+!!!example "Example cURL"
+    ```bash
+    curl --request POST \
+      --url 'https://management-api.experiment.amplitude.com/flags/<id>/variants/<variantKey>/users' \
```
Comment (bgiori) [inline docs/experiment/apis/management-api.md:1136]:
```suggestion
      --url 'https://management-api.experiment.amplitude.com/flags/<id>/variants/<variantKey>/users' \
      --header 'Content-Type: application/json' \
```

File: docs/experiment/apis/management-api.md [hunk 15]
```diff
@@ -478,13 +537,784 @@ Update the rollout weights for an experiment.
...
+### Response
+
+A successful request returns a `200 OK` response and a JSON object with the flag's details.
+
+<!-- TODO example response body -->
+
+!!!example "Example cURL"
+    ```bash
+    curl --request POST \
+      --url 'https://management-api.experiment.amplitude.com/flags/new' \
```
Comment (bgiori) [inline docs/experiment/apis/management-api.md:1283]:
```suggestion
      --url 'https://management-api.experiment.amplitude.com/flags/new' \
      --header 'Content-Type: application/json' \
```

File: docs/experiment/apis/management-api.md [hunk 16]
```diff
@@ -478,13 +537,784 @@ Update the rollout weights for an experiment.
...
+### Response
+
+A successful request returns a `200 OK` response.
+
+<!-- TODO example response body -->
+
+!!!example "Example cURL"
+    ```bash
+    curl --request PATCH \
+      --url 'https://management-api.experiment.amplitude.com/flags/<id>/variants/<variantKey>' \
```
Comment (bgiori) [inline docs/experiment/apis/management-api.md:1067]:
```suggestion
      --url 'https://management-api.experiment.amplitude.com/flags/<id>/variants/<variantKey>' \
      --header 'Content-Type: application/json' \
```

**Review by bgiori — CHANGES_REQUESTED** [review]

File: docs/experiment/apis/management-api.md (hunk 1 above)
Comment (imousterian) [reply docs/experiment/apis/management-api.md]: updated!

File: docs/experiment/apis/management-api.md (hunk 2 above)
Comment (imousterian) [reply docs/experiment/apis/management-api.md]: I provided better explanation

**Comment by imousterian** [comment]
@bgiori this is ready for re-review

File: docs/experiment/apis/management-api.md [hunk 17]
```diff
@@ -86,405 +128,1198 @@ A successful request returns a `200 OK` response and a list of experiments encod
...
+#### `filters`
+
+The `filters` field contains these objects.
+
+|<div class="med-big-column">Name</div>|Requirement|Type|Description|
+|---|---|---|---|
+|`group_type`| Optional | string | Group type of the filter; can be null. Currently we only use `USER` value. |
+|`subprop_key`| Required | string | Filter's key; can be null. |
+|`subprop_op`| Required | string | The [operation](#subprop_op) to use in this filter. |
+|`subprop_type`| Required | string | Type of filter; can be null. One of `USER`, `GROUP`, `EVENT`. |
```
Comment (bgiori) [inline docs/experiment/apis/management-api.md:581]: Can you clarify what the three values mean, and how they differ?

File: docs/experiment/apis/management-api.md [hunk 18]
```diff
@@ -86,405 +128,1198 @@ A successful request returns a `200 OK` response and a list of experiments encod
...
+|`subprop_value`| Required | string array | Array of values. |
+
+#### `subprop_op`
+
+An string value representing operations on a property value. Possible values are: `is`, `is not`, `contains`, `does not contain`, `less`, `less or equal`, `greater`, `greater or equal`, `glob match`, `glob does not match`
```
Comment (bgiori) [inline docs/experiment/apis/management-api.md:586]: This is kind of difficult to read, could you add these in a bullet list?

File: docs/experiment/apis/management-api.md [hunk 19]
```diff
@@ -86,405 +128,1198 @@ A successful request returns a `200 OK` response and a list of experiments encod
...
+|<div class="med-big-column">Name</div>|Requirement|Type|Description|
+|---|---|---|---|
+|`type`| Required | string | **Must have value: `property`** |
+|`prop`| Required | string | The property to use in the condition. |
+|`op`| Required | string | The [operation](#op) to use in this condition. |
+|`values`| Required | string array | The values to use in the operation. |
+
+#### `op`
+
+An string value representing operations on a property value. Possible values are: `is`, `is not`, `contains`, `does not contain`, `less`, `less or equal`, `greater`, `greater or equal`, `glob match`, `glob does not match`
```
Comment (bgiori) [inline docs/experiment/apis/management-api.md:665]: Bullet list

File: docs/experiment/apis/management-api.md [hunk 20]
```diff
@@ -86,405 +128,1198 @@ A successful request returns a `200 OK` response and a list of experiments encod
...
+      --header 'Accept: application/json' \
+      --header 'Authorization: Bearer <management-api-key>' \
+      --data '{"projectId":"<projectId>","key":"<key>"}'
+    ```
+
+------
+
+## Activate experiment
+
+!!!warn "Not recommended. Use [`edit experiment`](#edit-experiment) instead."
```
Comment (bgiori) [inline docs/experiment/apis/management-api.md:687]:
```suggestion
!!!warning "Not recommended. Use [`edit experiment`](#edit-experiment) instead."
```

Comment (bgiori) [reply docs/experiment/apis/management-api.md:687]: Got this callout name wrong

File: docs/experiment/apis/management-api.md [hunk 21]
```diff
@@ -86,405 +128,1198 @@ A successful request returns a `200 OK` response and a list of experiments encod
...
+      --header 'Content-Type: application/json' \
+      --header 'Accept: application/json' \
+      --header 'Authorization: Bearer <management-api-key>'
+    ```
+
+------
+
+## Deactivate experiment
+
+!!!warn "Not recommended. Use [`edit experiment`](#edit-experiment) instead."
```
Comment (bgiori) [inline docs/experiment/apis/management-api.md:716]:
```suggestion
!!!warning "Not recommended. Use [`edit experiment`](#edit-experiment) instead."
```

File: docs/experiment/apis/management-api.md [hunk 22]
```diff
@@ -86,405 +128,1198 @@ A successful request returns a `200 OK` response and a list of experiments encod
...
+      --header 'Content-Type: application/json' \
+      --header 'Accept: application/json' \
+      --header 'Authorization: Bearer <management-api-key>'
+    ```
+
+------
+
+## Rollout weights
+
+!!!warn "Not recommended. Use [`edit experiment variant`](#edit-experiment-variant) instead."
```
Comment (bgiori) [inline docs/experiment/apis/management-api.md:745]:
```suggestion
!!!warning "Not recommended. Use [`edit experiment variant`](#edit-experiment-variant) instead."
```

File: docs/experiment/apis/management-api.md [hunk 23]
```diff
@@ -86,405 +128,1198 @@ A successful request returns a `200 OK` response and a list of experiments encod
...
+A successful request returns a `200 OK` response.
+
+<!-- TODO example response body -->
+
+!!!example "Example cURL"
+    ```bash
+    curl --request PATCH \
+      --url 'https://management-api.experiment.amplitude.com/flags/<id>' \
+      --header 'Accept: application/json' \
+      --header 'Authorization: Bearer <management-api-key>'
```
Comment (bgiori) [inline docs/experiment/apis/management-api.md:1249]:
```suggestion
      --header 'Authorization: Bearer <management-api-key>' \
```

File: docs/experiment/apis/management-api.md [hunk 24]
```diff
@@ -86,405 +128,1198 @@ A successful request returns a `200 OK` response and a list of experiments encod
...
+### Response
+
+A successful request returns a `200 OK` response.
+
+<!-- TODO example response body -->
+
+!!!example "Example cURL"
+    ```bash
+    curl --request PATCH \
+      --url 'https://management-api.experiment.amplitude.com/flags/<id>' \
```
Comment (bgiori) [inline docs/experiment/apis/management-api.md:1247]:
```suggestion
      --url 'https://management-api.experiment.amplitude.com/flags/<id>' \
      --header 'Content-Type: application/json' \
```

**Review by bgiori — CHANGES_REQUESTED** [review]

**Review by bgiori — APPROVED** [review]
🏆🎉

**Pull request merged**